tests-pytest/action.yml

```diff
@@ -0,0 +1,87 @@
+name: "Run a test suite using the pytest framework"
+description: "Run a test suite using the pytest framework"
+
+inputs:
+  python-version:
+    description: 'Desired Python version for pytest'
+    default: '3.10'
+    required: false
+    type: string
+  pytest-markers:
+    description: 'Set of markers to filter out tests'
+    default: ''
+    required: false
+    type: string
+  pytest-extra-args:
+    description: 'Set of options to pass to the Sphinx builder'
+    default: ''
+    required: false
+  pytest-postargs:
+    description: 'Directory of the test suite and the level of verbosity'
+    default: 'tests -vv'
+    required: false
+    type: string
+  requires-xvfb:
+      description: 'Requires the usage of X Virtual Frame Buffer'
+      default: false
+      required: false
+      type: boolean
+
+runs:
+  using: "composite"
+  steps:
+  
+    - name: "Install Git and clone project"
+      uses: actions/checkout@v3
+  
+    - name: "Set up Python"
+      uses: actions/setup-python@v4.2.0
+      with:
+        python-version: ${{ inputs.python-version }}
+
+    - name: "Update pip"
+      shell: bash
+      run: python -m pip install -U pip
+  
+    - name: "Install Python library"
+      shell: bash
+      run:
+        python -m pip install .
+
+    - name: "Install X Virtual Frame Buffer"
+      shell: bash
+      if: inputs.requires-xvfb
+      run: |
+        sudo apt-get install xvfb
+
+    - name: "Check if requirements.txt file exists"
+      shell: bash
+      run: |
+        echo "EXISTS_TESTS_REQUIREMENTS=$(if [ -f requirements/requirements_tests.txt ]; then echo 'true'; else echo 'false'; fi)" >> $GITHUB_ENV
+
+    - name: "Print previous output"
+      shell: bash
+      run:
+        echo "Output was found ${{ env.EXISTS_TESTS_REQUIREMENTS }}"
+
+    - name: "Install documentation dependencies from requirements file"
+      shell: bash
+      if: env.EXISTS_TESTS_REQUIREMENTS == 'true'
+      run: python -m pip install -r requirements/requirements_tests.txt
+
+    - name: "Install documentation dependencies from pyproject.toml"
+      shell: bash
+      if: env.EXISTS_TESTS_REQUIREMENTS == 'false'
+      run: python -m pip install .[tests]
+    
+    - name: "Executing test suite without using xvfb"
+      if: inputs.requires-xvfb == 'false'
+      shell: bash
+      run: |
+        pytest ${{ inputs.pytest-markers }} ${{ inputs.pytest-extra-args }} ${{ inputs.pytest-postargs }}
+
+    - name: "Executing test suite using xvfb"
+      if: inputs.requires-xvfb == 'true'
+      shell: bash
+      run: |
+        xvfb-run pytest ${{ inputs.pytest-markers }} ${{ inputs.pytest-extra-args }} ${{ inputs.pytest-postargs }}
```

Comment:
Sometimes projects provide their own arguments in the pyproject.toml file

```toml
[tool.pytest.ini_options]
minversion = "7.1"
addopts = "-ra --cov=ansys.geometry --cov-report html:.cov/html --cov-report xml:.cov/xml --cov-report term -vv"
testpaths = [
    "tests",
]
```

If you provide input arguments I guess you would be overriding them, right?

Reply:
Yeah, we should specify it somewhere... If you just run "pytest" with no args, it will take the args from the pyproject.toml file or pytest.ini file. Otherwise, it will do whatever you tell the action to pass as arguments.

Maybe clarify this in the docs?